controllers/storagecluster/cephnfs.go

```diff
@@ -0,0 +1,274 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	"github.com/red-hat-storage/ocs-operator/controllers/defaults"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type ocsCephNFS struct{}
+
+// newCephNFSInstance returns the cephNFS instance that should be created
+// on first run.
+func (r *StorageClusterReconciler) newCephNFSInstances(initData *ocsv1.StorageCluster) ([]*cephv1.CephNFS, error) {
+	ret := []*cephv1.CephNFS{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generateNameForCephNFS(initData),
+				Namespace: initData.Namespace,
+			},
+			Spec: cephv1.NFSGaneshaSpec{
+				Server: cephv1.GaneshaServerSpec{
+					Active:    1,
+					Placement: getPlacement(initData, "nfs"),
+					Resources: defaults.GetDaemonResources("nfs", initData.Spec.Resources),
+					// set PriorityClassName for the NFS pods
+					PriorityClassName: openshiftUserCritical,
```

Comment:
a comment explaining _why_ it is `openshiftUserCritical` would be more useful than explaining that it gets set

I assume it is critical because Pods may use the NFS-volumes, and without it, they can not get started/stopped?

Reply:
done